Content.Server/Construction/Commands/TileWallsCommand.cs

```diff
@@ -104,19 +105,22 @@ public void Execute(IConsoleShell shell, string argStr, string[] args)
             var tile = mapSystem.GetTileRef(gridId.Value, grid, childTransform.Coordinates);
             var tileDef = (ContentTileDefinition)_tileDefManager[tile.Tile.TypeId];
 
-            if (tileDef.ID == TilePrototypeId)
+            if (tileDef.ID == TilePrototypeId && !tagSystem.HasTag(child, NaturalTag)) // Delta V - Begin add natural wall tile replace
             {
-                // Delta V - Begin add natural wall tile replace
-                if (tagSystem.HasTag(child, NaturalTag))
-                {
+                continue;
+            }
 
-                    mapSystem.SetTile(gridId.Value, grid, childTransform.Coordinates, naturalunderplatingTile);
-                    changed++;
+            if (tagSystem.HasTag(child, NaturalTag))
+            {
+                if (tileDef.ID == TilePrototypeId2)
+                {
                     continue;
                 }
-                // Delta V - end add natural wall tile replace
+                mapSystem.SetTile(gridId.Value, grid, childTransform.Coordinates, naturalunderplatingTile);
+                changed++;
                 continue;
             }
+            // Delta V - end add natural wall tile replace
```

Comment:
Missing `// Delta V - Begin add natural wall tile replace`

Reply:
I can move it up to 107 though if that's more clear?